CMakeLists.txt

```diff
@@ -209,7 +209,41 @@ include(CTest)
 enable_testing()
 
 if(GIT_FOUND AND ENABLE_GOOGLEMOCK)
-  add_subdirectory(gmock)
+  #add_subdirectory(gmock)
```

Comment:
can you describe which build problem you had? I can't reproduce them. 
I agree that introducing/using gmock in a project is not straight-forward but these changes look non obvious either.

Reply:
I got an error CMAKE_CXX_COMPILER not defined when adding googletest. Somehow the variable was not passed through properly to googletest. The code is mostly copy paste from the googletest docs: it downloads the source from git into the binary directory, runs cmake to configure an build and then uses the build binaries in your project. That seems to be the recommended way to include googletest in other projects. If you have specific questions, what is not clear, I can of course try to answer them.